### PR TITLE
[CLEAN-UP] Simplified saving and sharing

### PR DESCRIPTION
**Pull request by TiBiBa**
**Description**
In this PR we clean-up the structure for saving and sharing programs. Currently we only verify the correctness of programs when sharing one. But, why should we? We simplify our code by always parsing the code on save, independent of sharing or not. This way we reduce the function calls as we can always call the `saveit` function, only with an additional value for sharing or not.

**Fixes**
This fixes nothing, but cleans-up.

**How to test**
Make sure you are logged in and verify that all saving and sharing functionality is still working.

### CODE REVIEW

**Comment by TiBiBa** [comment]
@Felienne this one is ready for review!

File: website/programs.py [hunk 1]
```diff
@@ -69,22 +70,32 @@ def save_program(user):
...
         program_id = uuid.uuid4().hex
-        program_public = 0
+        program_public = body.get('shared')
         overwrite = False
         for program in programs:
             if program['name'] == body['name']:
                 overwrite = True
                 program_id = program['id']
-                program_public = program.get('public', 0)
+                program_public = program.get('public', program_public)
```
Comment (boryanagoncharenko) [inline website/programs.py:98]: This line means that I cannot share a program that was only saved but not shared. This is a deviation from the current behavior. Here are the steps:
- save a program (regardless if it is with errors or not)
- after the program is saved, click Save&Share and approve the override
- go to the my-programs page and check that the program is updated but not shared

**Review by boryanagoncharenko — CHANGES_REQUESTED** [review]

File: website/programs.py (hunk 1 above)
Comment (TiBiBa) [reply website/programs.py:98]: Will take a look!

Comment (TiBiBa) [reply website/programs.py]: Fixed!

**Comment by TiBiBa** [comment]
This should be ready for merging!

**Pull request merged**